spatial_cell_gpu.hpp

```diff
@@ -0,0 +1,1370 @@
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+/*!
+Spatial cell class for Vlasiator that supports a variable number of velocity blocks.
+*/
+
+#ifndef VLASIATOR_SPATIAL_CELL_GPU_HPP
+#define VLASIATOR_SPATIAL_CELL_GPU_HPP
+
+#include <algorithm>
+#include <cmath>
+#include <fstream>
+#include <iostream>
+#include <mpi.h>
+#include <limits>
+#include <stdint.h>
+#include <vector>
+#include <array>
+#include <unordered_map>
+#include <set>
+#include <map>
+#include <phiprof.hpp>
+#include <tuple>
+
+#include "memoryallocation.h"
+#include "common.h"
+#include "parameters.h"
+#include "definitions.h"
+
+#include "velocity_mesh_gpu.h"
+
+#include "velocity_block_container.h"
+
+#ifdef DEBUG_VLASIATOR
+   #ifndef DEBUG_SPATIAL_CELL
+   #define DEBUG_SPATIAL_CELL
+   #endif
+#endif
+
+typedef Parameters P; // Heeded in numerous files which include this one
+
+/*!
+Used as an error from functions returning velocity cells or
+as a cell that would be outside of the velocity block
+*/
+#define error_velocity_cell 0xFFFFFFFFu
+
+/*!
+Used as an error from functions returning velocity cell indices or
+as an index that would be outside of the velocity block
+*/
+#define error_velocity_cell_index 0xFFFFFFFFu
+
+namespace spatial_cell {
+
+   namespace Transfer {
+      const uint64_t NONE                     = 0;
+      const uint64_t CELL_PARAMETERS          = (1ull<<0);
+      const uint64_t CELL_DERIVATIVES         = (1ull<<1);
+      const uint64_t VEL_BLOCK_LIST_STAGE1    = (1ull<<2);
+      const uint64_t VEL_BLOCK_LIST_STAGE2    = (1ull<<3);
+      const uint64_t VEL_BLOCK_DATA           = (1ull<<4);
+      const uint64_t VEL_BLOCK_PARAMETERS     = (1ull<<6);
+      const uint64_t VEL_BLOCK_WITH_CONTENT_STAGE1  = (1ull<<7);
+      const uint64_t VEL_BLOCK_WITH_CONTENT_STAGE2  = (1ull<<8);
+      const uint64_t CELL_SYSBOUNDARYFLAG     = (1ull<<9);
+      const uint64_t CELL_E                   = (1ull<<10);
+      const uint64_t CELL_EDT2                = (1ull<<11);
+      const uint64_t CELL_PERB                = (1ull<<12);
+      const uint64_t CELL_PERBDT2             = (1ull<<13);
+      const uint64_t CELL_RHOM_V              = (1ull<<14);
+      const uint64_t CELL_RHOMDT2_VDT2        = (1ull<<15);
+      const uint64_t CELL_RHOQ                = (1ull<<16);
+      const uint64_t CELL_RHOQDT2             = (1ull<<17);
+      const uint64_t CELL_BVOL                = (1ull<<18);
+      const uint64_t CELL_BVOL_DERIVATIVES    = (1ull<<19);
+      const uint64_t CELL_DIMENSIONS          = (1ull<<20);
+      const uint64_t CELL_IOLOCALCELLID       = (1ull<<21);
+      const uint64_t NEIGHBOR_VEL_BLOCK_DATA  = (1ull<<22);
+      const uint64_t CELL_HALL_TERM           = (1ull<<23);
+      const uint64_t CELL_P                   = (1ull<<24);
+      const uint64_t CELL_PDT2                = (1ull<<25);
+      const uint64_t POP_METADATA             = (1ull<<26);
+      const uint64_t RANDOMGEN                = (1ull<<27);
```

Comment:
That seems dead, it's used in the ALL_DATA and ALL_SPATIAL_DATA below but not implemented. And I think it was dropped in the CPU version.